orchestration/templates/diff-table.yaml

```diff
@@ -0,0 +1,89 @@
+                - name: input-prefix
+                  value: '{{ "{{inputs.parameters.input-prefix}}" }}'
+                - name: staging-bq-project
+                  value: '{{ "{{inputs.parameters.staging-bq-project}}" }}'
+                - name: staging-bq-dataset
+                  value: '{{ "{{inputs.parameters.staging-bq-dataset}}" }}'
+                - name: jade-bq-project
+                  value: '{{ "{{inputs.parameters.jade-bq-project}}" }}'
+                - name: jade-bq-dataset
+                  value: '{{ "{{inputs.parameters.jade-bq-dataset}}" }}'
+
+    - name: join-staging-to-existing
+      inputs:
+        parameters:
+          - name: table-name
+          - name: gcs-bucket
+          - name: input-prefix
+          - name: staging-bq-project
+          - name: staging-bq-dataset
+          - name: jade-bq-project
+          - name: jade-bq-dataset
+      script:
+        image: us.gcr.io/broad-dsp-gcr-public/clinvar-schema:{{ default "latest" .Chart.AppVersion }}
+        env:
+          - name: STAGING_PROJECT
+            value: '{{ "{{inputs.parameters.staging-bq-project}}" }}'
+          - name: STAGING_DATASET
+            value: '{{ "{{inputs.parameters.staging-bq-dataset}}" }}'
+          - name: JADE_PROJECT
+            value: '{{ "{{inputs.paramaters.jade-bq-project}}" }}'
+          - name: JADE_DATASET
+            value: '{{ "{{inputs.parameters.jade-bq-dataset}}" }}'
+          - name: TABLE
+            value: '{{ "{{inputs.parameters.table-name}}" }}'
+          - name: GCS_PREFIX
+            value: '{{ "gs://{{inputs.parameters.gcs-bucket}}/{{inputs.parameters.input-prefix}}" }}'
+          - name: OUTPUT_SUFFIX
+            value: '{{ template "argo.timestamp" }}'
+        command: [bash]
+        source: |
+          set -euo pipefail
+
+          # Copy-pasted from StackOverflow.
+          function join_by { local d=$1; shift; echo -n "$1"; shift; printf "%s" "${@/#/$d}"; }
+
+          declare -r TABLE_DIR=/bq-metadata/${TABLE}
+          declare -r PK_COLS=$(cat ${TABLE_DIR}/primary-keys)
+          declare -r COMPARE_COLS=$(cat ${TABLE_DIR}/compare-cols)
+          declare -a COMPARISONS=()
+
+          for c in ${COMPARE_COLS//,/ }; do
+            COMPARISONS+=("TO_JSON_STRING(S.${c}) != TO_JSON_STRING(J.${c})")
+          done
+
+          declare -r FULL_DIFF=$(join_by ' OR ' "${COMPARISONS[@]}")
+
+          bq --location=US --project_id=${STAGING_PROJECT} --synchronous_mode=true --headless=true --format=none query \
+            --use_legacy_sql=false \
+            --external_table_definition=${TABLE}::${TABLE_DIR}/schema.json@NEWLINE_DELIMITED_JSON=${GCS_PREFIX}/${TABLE}/* \
+            --destination_table=${STAGING_PROJECT}:${STAGING_DATASET}.${TABLE}_${OUTPUT_SUFFIX} \
+            "SELECT J.datarepo_row_id, S.*
+             FROM ${TABLE} S FULL JOIN \`${JADE_PROJECT}.${JADE_DATASET}.${TABLE}\` J
+             USING (${PK_COLS}) WHERE ${FULL_DIFF}"
```

Comment:
Something I'm wondering about: Helm supports injecting arbitrary files into templates. We could pull all this bash out into a separate script, and inject it here. It'd be a little odd to look at the script in isolation, because all the env vars would be disconnected from their sources, but it might make this a little easier to read. WDYT?

Reply:
I haven't been looking at Argo workflows for too long but the huge block of bash does stick out in this file. I think separating it would be neater but if we frequently have to work with this logic, the var disconnect may be annoying.

Reply:
I think I'm leaning towards a separate bash script. Probably more readable.